libr/util/buf.c

```diff
@@ -461,15 +461,29 @@ static int r_buf_fcpy_at (RBuffer *b, ut64 addr, ut8 *buf, const char *fmt, int
 
 		for (k = 0; k < m; k++) {
 			if (write) {
-				r_mem_copyendian (
-				(ut8*)&buf[addr+len+(k*tsize)],
-				(ut8*)&b->buf[len+(k*tsize)],
-				tsize, endian);
+				if (bigendian /* XXX which endian? */) {
+					r_mem_swapendian (
```

Comment:
use ternary operator here to avoid duplicating the same paremeters twice, which makes the maintainance + review harder.